packages/lib/src/vis/shared/ViewportCenterer.tsx

```diff
@@ -0,0 +1,29 @@
+import { useFrame, useThree } from '@react-three/fiber';
+import { useEffect, useRef } from 'react';
+import type { Vector2 } from 'three';
+
+import { useAxisSystemContext } from './AxisSystemContext';
+import { useMoveCameraTo } from './hooks';
+
+function ViewportCenterer() {
+  const { dataToWorld, worldToData } = useAxisSystemContext();
+  const viewportCenter = useRef<Vector2>();
+  const { position } = useThree((state) => state.camera);
+
+  const moveCameraTo = useMoveCameraTo();
+
+  useFrame(() => {
+    viewportCenter.current = worldToData(position);
+  });
+
+  useEffect(() => {
+    if (viewportCenter.current) {
+      // On resize, move camera to the latest saved viewport center coordinates
+      const { x, y } = dataToWorld(viewportCenter.current);
```

Comment:
Note that I used the new utilities `worldToData` and `dataToWorld` rather than the scales. I find it easier to read.

Reply:
That's what it's for! :tada:

Reply:
Could `moveCameraTo` just accept a vector now?

Reply:
We could but two times out of three, we call it with a `Vector3`.

So we could have two possibility:
- Accept `Vector2 | Vector3` and discard the `z` component in the callback
- Accept only `Vector2` and create a new `Vector2` from the `Vector3` in the caller.

Both solutions are not very satisfactory to me. With `x, y`, we make clear that `z` is not changed.